elasticsearch/run.sh

```diff
@@ -28,4 +37,65 @@ else
 	exit 1
 fi
 
+
+# Wait for Elasticsearch port to be opened. Fail on timeout or if response from Elasticsearch is unexpected.
+wait_for_port_open() {
+	rm -f ${LOG_FILE}
+	echo -n "Checking if Elasticsearch is ready on ${ES_REST_BASEURL} "
+	while ! curl -s -k \
+			--cert ${secret_dir}admin-cert \
+			--key  ${secret_dir}admin-key \
+			--max-time ${max_time} \
+			-o ${LOG_FILE} \
+			${ES_REST_BASEURL} && [ ${timeouted} == false ]
+	do
+		echo -n "."
+		sleep ${RETRY_INTERVAL}
+		(( retry -= 1 ))
+		if (( retry == 0 )) ; then
+			timeouted=true
+		fi
+	done
+
+	# Test for response code 200 in Elasticsearch output. This can be sensitive to Elasticsearch version.
+	if [ -f ${LOG_FILE} ] && grep -q "200" ${LOG_FILE} ; then
+		echo "- connection successful"
+	else
+		if [ ${timeouted} == true ] ; then
+			echo -n "[timeout] "
+		fi
+		echo "failed"
+		cat ${LOG_FILE}
+		exit 1
+	fi
+}
+
+verify_or_add_index_template() {
+	wait_for_port_open
+	# Try to wait for cluster become more stable before index template being pushed in.
+	# Give up on timeout and continue...
+	curl -v -s -k -X GET \
+		--cert ${secret_dir}admin-cert \
+		--key  ${secret_dir}admin-key \
+		"${ES_REST_BASEURL}/_cluster/health?wait_for_status=yellow&timeout=${max_time}s"
+	# Check if index template already exists
+	response_code=$(curl -s -k -X HEAD \
+		--cert ${secret_dir}admin-cert \
+		--key  ${secret_dir}admin-key \
+		-w '%{response_code}' \
+		${ES_REST_BASEURL}/_template/viaq)
+	if [ $response_code == "200" ]; then
+		echo "Index template already present in ES cluster"
+	else
+		echo "Create index template"
+		curl -v -s -k -X PUT \
+			--cert ${secret_dir}admin-cert \
+			--key  ${secret_dir}admin-key \
+			-d@/usr/share/elasticsearch/config/com.redhat.viaq.template.json \
+			${ES_REST_BASEURL}/_template/viaq
```

Comment:
I think it is better to have a more descriptive name for the template in ES.
for example: `com.redhat.viaq-openshift`
Simple `viaq` can be used for a more generic template. or never.